### PR TITLE
fix: added initial provider authorize error rule

### PR DESCRIPTION
**Pull request by shanejonas**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26828?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. `CHAIN_PERMISSIONS=1 BARAD_DUR=1 yarn build:test`
2. `yarn test:api-specs-multichain`

## **Screenshots/Recordings**
image

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/lib/multichain-api/multichainMethodCallValidator.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const multichainMethodCallValidator = async (
     const result = v.validate(paramToCheck, p.schema as unknown as Schema, {
       required: true,
```
Comment (shanejonas) [inline app/scripts/lib/multichain-api/multichainMethodCallValidator.ts:66]:
```suggestion
      required: p.required,
```

File: app/scripts/lib/multichain-api/multichainMethodCallValidator.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export const multichainMethodCallValidator = async (
...
     });
-    if (result.errors) {
+    if (result.errors && p.required) {
```
Comment (shanejonas) [inline app/scripts/lib/multichain-api/multichainMethodCallValidator.ts:68]:
```suggestion
    if (result.errors) {
```

File: test/e2e/api-specs/MultichainAuthorizationConfirmationErrors.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+import _ from 'lodash';
+import { Driver } from '../webdriver/driver';
+import { WINDOW_TITLES, switchToOrOpenDapp } from '../helpers';
+import { addToQueue } from './helpers';
+
+type MultichainAuthorizationConfirmationOptions = {
+  driver: Driver;
+  only?: string[];
+};
+// this rule makes sure that a multichain authorization confirmation dialog is shown and confirmed
```
Comment (shanejonas) [inline test/e2e/api-specs/MultichainAuthorizationConfirmationErrors.ts:17]:
```suggestion
// this rule makes sure that a multichain authorization error codes are returned
```

File: test/e2e/api-specs/MultichainAuthorizationConfirmationErrors.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            } catch (e) {
+              console.log(e);
+            }
+          }
+        },
+      });
+    });
+  }
+
+  // get all the confirmation calls to make and expect to pass
```
Comment (shanejonas) [inline test/e2e/api-specs/MultichainAuthorizationConfirmationErrors.ts:71]:
```suggestion
```

File: test/e2e/api-specs/MultichainAuthorizationConfirmationErrors.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                  'eip155:1': {
+                    methods: ['eth_signTypedData_v4'],
+                    notifications: [],
+                  },
+                },
+                sessionProperties: {},
+              };
+              break;
+            default:
+              break;
```
Comment (jiexi) [inline test/e2e/api-specs/MultichainAuthorizationConfirmationErrors.ts:127]: do we need to do anything special if we don't match an error in this switch statement?

**Comment by shanejonas** [comment]
@metamaskbot update-policies

File: app/scripts/lib/multichain-api/multichainMethodCallValidator.ts [hunk 6]
```diff
@@ -55,20 +55,22 @@ export const multichainMethodCallValidator = async (
...
       paramToCheck = params[p.name];
-    } else {
+    } else if (params && Array.isArray(params)) {
       paramToCheck = params[i];
+    } else {
+      paramToCheck = undefined;
     }
     const result = v.validate(paramToCheck, p.schema as unknown as Schema, {
-      required: true,
+      required: p.required,
```
Comment (shanejonas) [inline app/scripts/lib/multichain-api/multichainMethodCallValidator.ts:68]: i made this change because it was making every param required.

**Comment by shanejonas** [comment]
@metamaskbot update-policies

**Review by jiexi — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**